src/main/java/org/opennars/interfaces/HasSensoryChannels.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.opennars.interfaces;
+
+import org.opennars.plugin.perception.SensoryChannel;
+
+/**
+ * Implementations have sensory channels
+ */
+public interface HasSensoryChannels {
```

Comment:
nethods that start with has is resered in java for Boolean accessors. should rename this.Perhaps `SensoryChannelConsumer`